syntax = "proto3";

option java_multiple_files = true;
package com.example.rpc.inventory;

// The request message containing the product ID and quantity needed
message InventoryRequest {
  string product_id = 1;
  int32 quantity = 2;
}

// The response message containing the status
message InventoryResponse {
  bool is_available = 1;
  string message = 2;
}

// Defining the service and the method
service InventoryService {
  rpc CheckStock(InventoryRequest) returns (InventoryResponse);
}